cognite/extractorutils/uploader.py

```diff
@@ -203,12 +285,29 @@ def __init__(
         max_upload_interval: Optional[int] = None,
         trigger_log_level: str = "DEBUG",
         thread_name: Optional[str] = None,
+        cancelation_token: threading.Event = threading.Event(),
     ):
         # Super sets post_upload and thresholds
         super().__init__(
-            cdf_client, post_upload_function, max_queue_size, max_upload_interval, trigger_log_level, thread_name
+            cdf_client,
+            post_upload_function,
+            max_queue_size,
+            max_upload_interval,
+            trigger_log_level,
+            thread_name,
+            cancelation_token,
         )
-        self.upload_queue: Dict[str, Dict[str, List[Row]]] = dict()
+        self.upload_queue: Dict[str, Dict[str, List[TimestampedObject]]] = dict()
+
+        # It is a hack since Prometheus client registers metrics on object creation, so object has to be created once
+        # Store the reference to a global objects.
+        global raw_uploader_rows_queued, raw_uploader_rows_written, raw_uploader_rows_duplicates
+        global raw_uploader_queue_size, raw_uploader_latency
+        self.rows_queued = raw_uploader_rows_queued
```

Comment:
Why add these as instance attributes, and not just use the global variables directly in the other methods?

Reply:
I was thinking to make it clear that those come outside, but strictly speaking, we can rename and drop the global declaration

Reply:
I was thinking more of why we add `self.rows_queued` instead of just using `raw_uploader_rows_queued` directly

Reply:
If you think it's cleaner, that's fine 🙂 I was just wondering what the reasoning was.

Reply:
Nevermind, seeing the new changes after the renaming changed my mind 🙂 let's keep them